muskrat/s3consumer.py

```diff
@@ -9,10 +9,54 @@
 """
 import os
 import time
+import json
 
-import boto
+import boto3
 from   muskrat.util import config_loader
 
+
+def s3collection_marker_each_cur(collection, cursor):
+    """
+    Helper for iterating through s3 objects using a persistent marker. Loads
+    the marker from the given file when present, and creates it if not present.
+
+    collection
+        a collection of boto3 s3 ObjectSummary resources
+    cursor
+        S3Cursor instance where the marker will be read and stored
+    """
+    marker = cursor.get()
+    if marker:
+        collection = collection.filter(Marker=marker)
+
+    for obj in collection:
+        yield obj
+        cursor.update(obj.key)
+
+
+def s3collection_marker_each(collection, path):
+    """
+    Helper for iterating through s3 objects using a persistent marker. Loads
+    the marker from the given file when present, and creates it if not present.
+
+    collection
+        a collection of boto3 s3 ObjectSummary resources
+    path
+        path to the persistent marker file
+    """
+    name = os.path.basename(path)
+    cursor = S3Cursor(name, 'file', location=os.path.dirname(path))
+    return s3collection_marker_each_cur(collection, cursor)
+
+
+def s3collection_marker_json(collection, path):
```

Comment:
Probably good to be able to set a property via the S3Consumer class + decorator that allows use to elect to feed the consumer a string (default ... cause everything is treating this as a string already) or json.  Would be nice to write new consumers using json.